colcon_in_container/verb/build_in_container.py

```diff
@@ -119,12 +119,15 @@ def main(self, *, context):  # noqa: D102
                 self.provider.upload_package(package.path)
             exit_code = self._build(context.args)
 
-        if exit_code and context.args.debug:
-            logger.error(f'Build failed with error code {exit_code}.')
-            logger.warn('Debug was selected, entering the instance.')
-            self.provider.shell()
+        if exit_code != 0:
+            logger.error(f'Build failed with exit code {exit_code}. ')
+            if context.args.debug:
+                logger.warn('Debug was selected, entering the instance.')
+                self.provider.shell()
+            sys.exit(1)
         elif context.args.shell_after:
```

Comment:
There is a behaviour change here.
the `shell_after` option is meant to shell into the container in any case.
Meaning, whether there is an error or not.
With this change, in case of an error (and `debug` was not used), we won't shell into the container while this is what we want.